File: backend/models/invoice.py
from enum import Enum
from pydantic import UUID4
from .base import TimestampSchema

class InvoiceStatus(str, Enum):
    """Enumeration of invoice statuses."""
    pending = 'pending'
    paid = 'paid'
    overdue = 'overdue'
    cancelled = 'cancelled'

class InvoiceBase(TimestampSchema):
    """Base schema for Invoice data."""
    firm_id: UUID4
    amount: float
    status: InvoiceStatus

class InvoiceCreate(InvoiceBase):
    """Schema for creating a new Invoice."""
    pass

class Invoice(InvoiceBase):
    """Schema for representing an Invoice (includes ID)."""
    id: UUID4

File: frontend/src/interfaces/invoice.ts
import { BaseEntity } from './base';

/**
 * Enumeration of possible invoice statuses.
 */
export enum InvoiceStatus {
  PENDING = 'pending',
  PAID = 'paid',
  OVERDUE = 'overdue',
  CANCELLED = 'cancelled'
}

/**
 * Represents a Billing Invoice.
 */
export interface Invoice extends BaseEntity {
  firm_id: string;
  amount: number;
  status: InvoiceStatus;
}
